### PR TITLE
Tiffany/payment payslip

### PR DESCRIPTION
**Pull request by cnnmon**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

- Front end of the financial summary/payslip screen
- Takes in financial record as a prop & takes/displays amount billed
- Button to go to camera screen

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## How to review

http://localhost:3000/confirm-payment
Doesn't take in the right props currently as the screen leading into it (?) isn't built out yet.

[//]: # 'The order in which to review files and what to expect when testing locally'

## Next steps

- Camera functionality
- Passing down the right props
- Sending data to Airtable

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

### Screenshots

image

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @julianrkung

[//]: # 'This tags Julian as a default. Feel free to change, or add on anyone who you should be in on the conversation.'

### CODE REVIEW

File: src/screens/FinancialSummary/ConfirmPayment.tsx
Comment (julianrkung) [inline src/screens/FinancialSummary/ConfirmPayment.tsx]: NIT
```
border: `3.5px dashed ${theme.palette.divider}`,
```
template literal here is cleaner to read

File: src/screens/FinancialSummary/components/FinancialInfo.tsx
Comment (julianrkung) [inline src/screens/FinancialSummary/components/FinancialInfo.tsx]: `FinancialInfoProps`

File: src/screens/FinancialSummary/components/PaymentCard.tsx
Comment (julianrkung) [inline src/screens/FinancialSummary/components/PaymentCard.tsx]: singular since it's 1 button

File: src/screens/FinancialSummary/components/PaymentCard.tsx
Comment (julianrkung) [inline src/screens/FinancialSummary/components/PaymentCard.tsx]: Is this component used anywhere? I can't match this with anything on the [current Figma ](https://www.figma.com/file/MzpMtoYyVYFn19DpAXHEci/Mee-Panyar?node-id=2161%3A2).

If it's not used anywhere, we might wanna not have this in the PR. A good rule of thumb is that we only PR things that are being used and are testable by the reviewer

File: src/screens/FinancialSummary/components/FinancialInfo.tsx
Comment (julianrkung) [inline src/screens/FinancialSummary/components/FinancialInfo.tsx]: Remove unused import

File: src/screens/FinancialSummary/ConfirmPayment.tsx
Comment (julianrkung) [inline src/screens/FinancialSummary/ConfirmPayment.tsx]: Remove unused import

File: src/App.tsx
Comment (julianrkung) [inline src/App.tsx]: We'll want to change this route once we've built out the remainder of the Financial Summary screens. It'll eventually look something like `/financial-summaries/financial-summary/payment/create` imo. Can we add a TODO above this to change the route name once the other screens are built out?

File: src/screens/FinancialSummary/ConfirmPayment.tsx
Comment (julianrkung) [inline src/screens/FinancialSummary/ConfirmPayment.tsx]: Name isn't descriptive enough. It doesn't tell enough about what this component is (ideally, i'd be able to match a specific component like this one to the screen it's on just by looking at it's name). For components especially, longer names are better (unless it's egregiously long). I wouldn't mind something along the lines of `FinancialSummaryPayment` (which isn't really perfect either)

**Review by julianrkung — COMMENTED** [review]
@cnnmon To make it easier for reviewers (so it doesn't error), could we for now just use the first Financial Summary when we go to the route (also add a TODO to remove this once the screen that leads into this is built out)? That's my bad since for some reason I thought Maria had built out the component that links into this one the semester before.

I added the following to `ConfirmPayment` (and imported the functions at the top) as a quick hack to get it to run without error FYI @wangannie when you review.

```
const mapStateToProps = (state: RootState) => ({
  financialSummary: state.siteData.currentSite.financialSummaries[0] || null,
});

export default connect(mapStateToProps)(withStyles(styles)(ConfirmPayment));

```

File: src/screens/FinancialSummary/components/PaymentCard.tsx
Comment (wangannie) [inline src/screens/FinancialSummary/components/PaymentCard.tsx]: Rename to be more specific

File: src/screens/FinancialSummary/ConfirmPayment.tsx
Comment (wangannie) [inline src/screens/FinancialSummary/ConfirmPayment.tsx]: Nit to match Figma: this is supposed to fit flush with the rest of the text labels, so you don't need this margin here. Just for best practice, I'd also cut the `width` here and instead add this to the `content` container:
```
display: 'flex',
flexDirection: 'column',
```

Comment (wangannie) [reply src/screens/FinancialSummary/ConfirmPayment.tsx]: I see there's also a 10 margin in `FinancialInfo` -- I'd also take that out to make it flush.

File: src/screens/FinancialSummary/ConfirmPayment.tsx
Comment (wangannie) [inline src/screens/FinancialSummary/ConfirmPayment.tsx]: For the sake of consistency, you should probably wrap this in a Typography tag. Then you don't have to use the `<br />` and you can use flexbox to put these in a centered column layout!

File: src/screens/FinancialSummary/components/FinancialInfo.tsx
Comment (wangannie) [inline src/screens/FinancialSummary/components/FinancialInfo.tsx:30]: Looks like this should be red according to Figma

File: src/screens/FinancialSummary/components/FinancialInfo.tsx
Comment (wangannie) [inline src/screens/FinancialSummary/components/FinancialInfo.tsx]: Is this ****** temporary? I'm curious how sensitive this number is/how we should handle this.

**Review by wangannie — COMMENTED** [review]
I didn't dive too deep into the CSS but I left some higher-level styling comments. @julianrkung mentioned styling isn't the highest priority right now, so defer to him on if you should address/ignore my comments. Otherwise, this looks great! A+

File: src/screens/FinancialSummary/components/FinancialInfo.tsx
Comment (julianrkung) [reply src/screens/FinancialSummary/components/FinancialInfo.tsx]: True, I think on the Adalo app it showed the entire accounting number (so it might be easier to just do that). I don't think it's a big deal either way, though.

File: src/screens/FinancialSummary/FinancialSummaryPayment.tsx
Comment (julianrkung) [inline src/screens/FinancialSummary/FinancialSummaryPayment.tsx]: Is there a reason this TODO can't be done right now? Looks like we can just pass the `EMPTY_SUMMARY` if we don't have a summary in redux.

File: src/lib/redux/siteDataSlice.ts
Comment (julianrkung) [inline src/lib/redux/siteDataSlice.ts:30]: add `id: '',` to top of this since `FinancialSummaryRecord` now has ids!

image

**Review by julianrkung — APPROVED** [review]
LGTM after you add the `id` field to `EMPTY_FINANCIAL_SUMMARY`.

Good work :) high quality PR!

**Pull request merged**
